data-service/src/test/java/com/amazonaws/blox/dataservice/repository/EnvironmentRepositoryDDBTest.java

```diff
@@ -198,4 +255,197 @@ public void geEnvironment() throws StorageException, EnvironmentNotFoundExceptio
+    environmentRepositoryDDB.getEnvironmentTargetVersion(null);
+  }
+
+  @Test(expected = StorageException.class)
+  public void getEnvironmentTargetVersionStorageException() throws StorageException {
+    doThrow(new AmazonServiceException(""))
+        .when(dynamoDBMapper)
+        .load(isA(EnvironmentTargetVersionDDBRecord.class));
+    environmentRepositoryDDB.getEnvironmentTargetVersion(environmentVersion.getEnvironmentId());
+  }
+
+  @Test
+  public void getEnvironmentTargetVersion() throws StorageException {
+    when(dynamoDBMapper.load(isA(EnvironmentTargetVersionDDBRecord.class)))
+        .thenReturn(environmentTargetVersionDDBRecord);
+
+    final EnvironmentVersion environmentVersionResult =
+        environmentRepositoryDDB.getEnvironmentTargetVersion(environmentVersion.getEnvironmentId());
+    assertEquals(environmentVersion, environmentVersionResult);
+
+    verify(dynamoDBMapper)
+        .load(EnvironmentTargetVersionDDBRecord.withHashKey(environmentVersion.getEnvironmentId()));
+  }
+
+  @Test(expected = StorageException.class)
+  public void listClustersStorageException() throws StorageException {
+    when(dynamoDBMapper.scan(
+            eq(EnvironmentTargetVersionDDBRecord.class), isA(DynamoDBScanExpression.class)))
+        .thenThrow(new AmazonServiceException(""));
+
+    environmentRepositoryDDB.listClusters();
+  }
+
+  @Test
+  public void listClusters() throws StorageException {
+    List<EnvironmentTargetVersionDDBRecord> environmentTargetVersionDDBRecords = new ArrayList<>();
+    environmentTargetVersionDDBRecords.add(environmentTargetVersionDDBRecord);
+    final EnvironmentTargetVersionDDBRecord secondRecord =
+        EnvironmentTargetVersionDDBRecord.builder()
+            .environmentId(SECOND_ENVIRONMENT_ID)
+            .environmentVersion(UUID.randomUUID().toString())
+            .cluster(SECOND_CLUSTER)
+            .build();
+    environmentTargetVersionDDBRecords.add(secondRecord);
+
+    final EnvironmentTargetVersionDDBRecord thirdRecordSameCluster =
+        EnvironmentTargetVersionDDBRecord.builder()
+            .environmentId(SECOND_ENVIRONMENT_ID)
+            .environmentVersion(UUID.randomUUID().toString())
+            .cluster(SECOND_CLUSTER)
+            .build();
+    environmentTargetVersionDDBRecords.add(thirdRecordSameCluster);
+
+    when(environmentTargetVersionDDBRecordPaginatedScanList.stream())
+        .thenReturn(environmentTargetVersionDDBRecords.stream());
+
+    when(dynamoDBMapper.scan(
+            eq(EnvironmentTargetVersionDDBRecord.class), isA(DynamoDBScanExpression.class)))
+        .thenReturn(environmentTargetVersionDDBRecordPaginatedScanList);
+
+    final List<String> clusters = environmentRepositoryDDB.listClusters();
+
+    final Set<String> expectedClusters = new HashSet<>();
+    for (EnvironmentTargetVersionDDBRecord record : environmentTargetVersionDDBRecords) {
+      expectedClusters.add(record.getCluster());
```

Comment:
nit: Is this really necessary? AFAICT `expectedClusters` is a two-element list with `FIRST_CLUSTER, SECOND_CLUSTER` in it...

Reply:
This is to generalize it and not depend on previous knowledge of arguments. I can use the hamcrest method you recommend below